balancer_conn_wrappers.go

```diff
@@ -359,8 +362,9 @@ func (ccb *ccBalancerWrapper) Target() string {
 // acBalancerWrapper is a wrapper on top of ac for balancers.
 // It implements balancer.SubConn interface.
 type acBalancerWrapper struct {
-	mu sync.Mutex
-	ac *addrConn
+	mu        sync.Mutex
+	ac        *addrConn
+	producers map[balancer.ProducerBuilder]*producerData
```

Comment:
I probably will have to make a couple of more passes at least to follow the code more closely.

Why do we not have a `Name()` method on the `Producer` interface to uniquely identify a producer? 
Is there going to be a valid use case where can have multiple instances of the same type of producer (so, different pointer each time) on a subConn?
Can/should this map be indexed on the name of the producer?

Reply:
> Why do we not have a Name() method on the Producer interface to uniquely identify a producer?

IMO that's more fragile, as it is possible for there to be multiple implementations of those builders, and they are not stored in a global registry (like how resolvers/balancers work).

> Is there going to be a valid use case where can have multiple instances of the same type of producer (so, different pointer each time) on a subConn?

This sort of defeats the purpose of sharing the producer.  I think this API is appropriate as-is, and a `Name()` method would be worse, as collisions would become possible.